provider/aws/registries.go

```diff
@@ -36,6 +36,17 @@ func (p *AWSProvider) RegistryAdd(server, username, password string) (*structs.R
 	// validate login
 	switch {
 	case regexpECRHost.MatchString(server):
+		system, err := p.describeStack(p.Rack)
+		if err != nil {
+			return nil, err
+		}
+		sid := regexpStackID.FindStringSubmatch(*system.StackId)[2]
+		host := fmt.Sprintf("%s.dkr.ecr.%s.amazonaws.com", sid, p.Region)
```

Comment:
This diverges from how the build process determines the app's registry. This seems in danger to diverge. Should fix one or both sides or at least add clarifying comments to both sides.